### PR TITLE
Get customer names by merchant

### PR DESCRIPTION
**Pull request by KMPWeiland**
Add customers_by_merchant Endpoint
*NB: this also includes an index action for customers, which is not required by the project spec.

Type of change

  - [x] New feature
  - [ ] Bug Fix

Check the correct boxes

  - [x] This code broke nothing
  - [ ] This code broke some stuff
  - [ ] This code broke everything

Implements/Fixes:

  - This PR implements the customers_by_merchant endpoint, allowing users to retrieve a list of customers associated with a specific merchant. It formats the response to include customer details (id, first_name, last_name) in JSON:API format.  The endpoint also handles edge cases where the merchant ID is not found (returns a 404 Not Found response).
  - Thie PR also implements the index action, allowing users to retrieve all customers, across all merchants. 
  - Testing is including in this PR, with 97% coverage.

Testing Changes:

  - [x] I have fully tested my code 
  - [x] All tests are passing
  - [ ] Some tests are failing
  - [ ] All tests are failing

  - [x] No previous tests have been changed
  - [ ] Some previous tests have been changed
  - [ ] All of the previous tests have been changed (Please describe what in the world happened that all of the previous tests needed changing.)

Checklist:

  - [x] I have reviewed my code
  - [x] The code will run locally
  - [x] My code has no unused/commented out code
  - [ ] I have commented my code, particularly in hard-to-understand areas

(Optional) What questions do you have? Anything specific you want feedback on?

  * The rspec file could be DRYer.  Refactor opp = using a before_each block for testing data.
  * The file system is not completely consistent with the team's. I could put the customers_spec file in a new customers directory, and even create two seperate files for the index test and customers_by_merchant test to match the file structure for the other tests.

Link: [get_customer_names_by_merchant](https://github.com/pshim17/little_shop/tree/get_customer_names_by_merchant)

### CODE REVIEW

File: app/controllers/api/v1/merchants/customers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class Api::V1::Merchants::CustomersController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :item_not_found
```
Comment (pshim17) [inline app/controllers/api/v1/merchants/customers_controller.rb:2]: Nice usage of rescue_from!

File: spec/customers_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
+require 'rails_helper'
```
Comment (pshim17) [inline spec/customers_controller_spec.rb:1]: Good job with the tests! It is clean, organized and easy to follow. Good job covering the happy paths/edge cases

**Review by pshim17 — APPROVED** [review]

**Comment by kevin-newland** [comment]
Great job! Everything is clear and organized!
